.travis.yml

```diff
@@ -15,12 +13,18 @@ python:
 
 matrix:
     include:
-        - python: 3.6
+        # Enable 3.7 without globally enabling sudo and dist: xenial for other build jobs
+        - python: 3.7
+          dist: xenial
```

Comment:
As xenial supports all Python environments, you can put xenial at the top level and avoid specifying it at every level of the `include:` block.

Reply:
Yes, some packages didn't want to move over to xenial for all builds.  If you'd prefer moving to xenial for all builds I can simplify this PR.

Reply:
IMO, a simpler configuration is preferable.

Reply:
It would appear that pypy and pypy3 aren't fully supported on xenial - https://travis-ci.community/t/confusing-pypy-versions/2505 - would you prefer to drop those builds or specify the exact versions of them as some of the comments suggest may work?

Reply:
Ah, right. They are under the aliases:

- pypy2.7-6.0
- pypy3.5-6.0

Try those instead. Don't forget to add "3.7" to the list of pythons.

Reply:
Looks like we're good now, might want to squash and merge this PR as there's quite a few commits for ultimately a small change.